docs/source/routing/performance/caching/response-caching/cache-expiration.mdx

```diff
@@ -0,0 +1,528 @@
+
+#### Add the directive to your schema
+
+First, add the `@cacheControl` directive definition to your subgraph schema:
+
+```graphql
+enum CacheControlScope {
+  PUBLIC
+  PRIVATE
+}
+
+directive @cacheControl(
+  maxAge: Int
+  scope: CacheControlScope
+  inheritMaxAge: Boolean
+) on FIELD_DEFINITION | OBJECT | INTERFACE | UNION
+```
+
+<Note>
+
+Apollo Server recognizes this directive automatically. For other GraphQL servers, consult your server's documentation to determine if it supports `@cacheControl` or an equivalent mechanism for setting `Cache-Control` headers.
+
+</Note>
+
+#### Field-level TTLs
+
+Apply `@cacheControl` to individual fields to set specific TTLs:
+
+```graphql
+type Post {
+  id: ID!
+  title: String
+  author: Author
+  votes: Int @cacheControl(maxAge: 30)
+  readByCurrentUser: Boolean! @cacheControl(maxAge: 10, scope: PRIVATE)
+}
+```
+
+In this example:
+- The `votes` field caches for 30 seconds
+- The `readByCurrentUser` field caches for 10 seconds and is marked as private (user-specific)
+
+#### Type-level TTLs
+
+Apply `@cacheControl` to types to set a default TTL for all fields returning that type:
+
+```graphql
+type Post @cacheControl(maxAge: 240) {
+  id: Int!
+  title: String
+  author: Author
+  votes: Int
+}
+
+type Comment {
+  post: Post!  # Cached for up to 240 seconds
+  body: String!
+}
+```
+
+Field-level settings override type-level settings:
+
+```graphql
+type Comment {
+  post: Post! @cacheControl(maxAge: 120)  # Overrides the 240s type-level setting
+  body: String!
+}
+```
+
+#### Scope: PUBLIC vs PRIVATE
+
+The `scope` argument controls whether cached data can be shared across users:
+
+- `PUBLIC` (default): The data is identical for all users and can be shared in the cache
+- `PRIVATE`: The data is user-specific and requires a `private_id` configuration to cache per-user
+
+```graphql
+type User {
+  id: ID!
+  name: String @cacheControl(maxAge: 3600)  # Public: same for all users
+  cart: [Product!]! @cacheControl(maxAge: 60, scope: PRIVATE)  # Private: per-user
+}
+```
+
+See the [Customization](/router/performance/caching/response-caching/customization#private-data-caching) page for details on caching private data.
+
+#### Complete example
+
+This schema example uses both `@cacheTag` (for invalidation) and `@cacheControl` (for TTL):
+
+```graphql
+extend schema
+  @link(
+    url: "https://specs.apollo.dev/federation/v2.12"
+    import: ["@key", "@requires", "@external", "@cacheTag"]
+  )
+
+enum CacheControlScope {
+  PUBLIC
+  PRIVATE
+}
+
+directive @cacheControl(
+  maxAge: Int
+  scope: CacheControlScope
+  inheritMaxAge: Boolean
+) on FIELD_DEFINITION | OBJECT | INTERFACE | UNION
+
+type Query {
+  user(id: ID!): User
+  users: [User!]!
+}
+
+type User @key(fields: "id") @cacheTag(format: "user-{$key.id}") @cacheControl(maxAge: 10) {
```

Comment:
I don't think we should talk about cacheTag here because we introduce active invalidation just after